src/Stratis.Bitcoin.Features.SmartContracts.Tests/Receipts/ReceiptSerializationTest.cs

```diff
@@ -70,24 +75,32 @@ private void TestStorageSerialize(Receipt receipt)
         {
             byte[] serialized = receipt.ToStorageBytesRlp();
             Receipt deserialized = Receipt.FromStorageBytesRlp(serialized);
-            Assert.Equal(receipt.PostState, deserialized.PostState);
-            Assert.Equal(receipt.GasUsed, deserialized.GasUsed);
-            Assert.Equal(receipt.Bloom, deserialized.Bloom);
-            Assert.Equal(receipt.Logs.Length, deserialized.Logs.Length);
+            TestStorageReceiptEquality(receipt, deserialized);
+        }
 
-            for (int i = 0; i < receipt.Logs.Length; i++)
+        /// <summary>
+        /// Ensures 2 receipts and all their properties are equal.
+        /// </summary>
+        public static void TestStorageReceiptEquality(Receipt receipt1, Receipt receipt2)
+        {
```

Comment:
This functionality could be on `Receipt`.

Reply:
Hmm we could, I don't think we will ever need to do that in bizlogic though, it is probably more useful to keep equality checks doing a reference check